### PR TITLE
Fix python plugins, add CLBOSS, and fix port to JS

### PR DESCRIPTION
**Pull request by chrisguida**
This PR fixes everything

- Python plugins (broken in last release):
  - rebalance
  - summary

- Builds in a sensible way due to using debian instead of alpine
- Adds CLBOSS plugin
- Now builds on Mac
- Produces a much smaller image than before, not counting the CLBOSS binary

TODO: 
- [ ] ~~the binary for CLBOSS is currently 350 MB, need to shrink~~ binary is huge, hopefully can be solved upstream (https://github.com/ZmnSCPxj/clboss/issues/138)
- [x] Fix build, currently we manually need to `git clone --no-checkout` in order to get the `.git` dir from the submodule repo, since the build is not expecting a submodule

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,108 +1,159 @@
...
+VOLUME [ "/root/.lightning" ]
+COPY --from=builder /tmp/lightning_install/ /usr/local/
+COPY --from=downloader /opt/bitcoin/bin /usr/bin
 
 RUN wget https://github.com/mikefarah/yq/releases/download/v4.12.2/yq_linux_arm.tar.gz -O - |\
     tar xz && mv yq_linux_arm /usr/bin/yq
 
-COPY --from=builder /usr/local /usr/local
-COPY --from=bitcoin-core /bitcoin-${BITCOIN_VERSION}/src/bitcoin-cli /usr/local/bin
+# COPY --from=builder /usr/local /usr/local
```
Comment (Blu-J) [inline Dockerfile:139]: Do we want this uncommented out? still

File: Dockerfile [hunk 2]
```diff
@@ -1,108 +1,159 @@
...
-
-# Builder
-FROM alpine:3.12 as builder
-
-RUN apk add ca-certificates alpine-sdk autoconf automake git libtool gmp-dev \
-    sqlite-dev python2 python3 py3-pip py3-mako net-tools zlib-dev libsodium gettext jq
-RUN pip3 install mrkd mistune==0.8.4
-
-ADD ./.gitmodules /root/.gitmodules
-ADD ./.git /root/.git
```
Comment (Blu-J) [inline Dockerfile:64]: :speech_balloon: Thanks for removing this, this was getting on my sanity.

File: Makefile [hunk 3]
```diff
@@ -23,12 +23,19 @@ c-lightning.s9pk: manifest.yaml assets/compat/* image.tar instructions.md $(ASSE
 	embassy-sdk pack
 
 image.tar: Dockerfile docker_entrypoint.sh check-rpc.sh check-synced.sh configurator/target/aarch64-unknown-linux-musl/release/configurator c-lightning-http-plugin/target/aarch64-unknown-linux-musl/release/c-lightning-http-plugin $(C_LIGHTNING_GIT_FILE) $(PLUGINS_SRC) $(C_LIGHTNING_REST_SRC) manifest.yaml
+# dumb hack to give the c-lightning build a .git folder because it insists on breaking on a submodule .git file
```
Comment (Blu-J) [inline Makefile:26]: :speech_balloon: Thanks for indicating that this was a hack. This was better than the last hack, so boyscout rule says ok. Can't get perfect, and can't think of a perfect. I'm good with this hack

File: docker_entrypoint.sh [hunk 4]
```diff
@@ -40,12 +48,13 @@ do
     sleep 1
 done
 cp /usr/local/libexec/c-lightning/plugins/c-lightning-REST/certs/access.macaroon /root/.lightning/public/access.macaroon
-cat /root/.lightning/public/access.macaroon | base64  > /root/.lightning/start9/access.macaroon.base64
-cat /root/.lightning/public/access.macaroon | base64 | xxd  > /root/.lightning/start9/access.macaroon.hex
+cat /root/.lightning/public/access.macaroon | basenc --base64url -w0  > /root/.lightning/start9/access.macaroon.base64
```
Comment (Blu-J) [inline docker_entrypoint.sh:51]: :speech_balloon: Thanks for fixing, in a side conversation learned.
> they were producing extra output
the base64 you had put whitespace, and also was not url safe
and the hex had a whole bunch of formatting to see the bytes and their ascii encodings

File: docker_entrypoint.sh [hunk 5]
```diff
@@ -40,12 +48,13 @@ do
...
+cat /root/.lightning/public/access.macaroon | basenc --base16 -w0  > /root/.lightning/start9/access.macaroon.hex
 
 lightning-cli getinfo > /root/.lightning/start9/lightningGetInfo
 
 echo "All configuration Done"
 
+trap _term TERM
```
Comment (Blu-J) [inline docker_entrypoint.sh:58]: :speech_balloon: I'm going to have to remember this one.

File: Dockerfile (hunk 2 above)
Comment (Blu-J) [reply Dockerfile:64]: Ah, I learn it still doesn't fix.

File: Makefile (hunk 3 above)
Comment (chrisguida) [reply Makefile:26]: Sorry, the build still re-executes every time. I can't figure out how to make it use the docker cache.

File: Dockerfile (hunk 2 above)
Comment (chrisguida) [reply Dockerfile:64]: Yeah, sorry, the build still re-executes every time. I can't figure out how to make it use the docker cache.

File: Dockerfile (hunk 1 above)
Comment (chrisguida) [reply Dockerfile:139]: nah, removed

**Review by Blu-J — APPROVED** [review]

**Comment by chrisguida** [comment]
Proof:

Rebalance works now:
Screen Shot 2022-06-13 at 5 54 03 PM

Summary works now:
Screen Shot 2022-06-13 at 5 54 18 PM

Mac builds work now (on M1):
Screen Shot 2022-06-13 at 5 55 02 PM

CLBOSS added:
Screen Shot 2022-06-13 at 5 55 23 PM

SIGTERM works now:
Screen Shot 2022-06-13 at 6 00 37 PM

**Pull request merged**